### ISSUE TITLE
Crash when compiling with nims files - only on v0.19.2

### ISSUE THREAD

**Issue by genotrance** [nim-lang/Nim#10344]
Consider three files:

```nim
# a.nim
echo "Hello"
```

```nim
# config.nims
switch("d", "ABC")
```

```nim
# a.nims
switch("d", "DEF")
```

When compiling `a.nim`, nim crashes. This is only seen on 0.19.2, not on 0.19.0 or on devel.

```
> nim c a.nim
Hint: used config file 'C:\Users\gt\.choosenim\toolchains\nim-0.19.2\config\nim.cfg' [Conf]
Hint: used config file 'C:\Dropbox\Private\github\nimterop\config.nims' [Conf]
???(0, 0) Error: internal error: n is not nil
No stack traceback available
To create a stacktrace, rerun compilation with ./koch temp c <file>
```

Delete one of the nims files and it works again.

I'm using `config.nims` for global flags and `a.nims` for a define that only applies in compiled executable.

**Comment by Araq** [comment]
@timotheecour is the bugfix/commit marked with ``[backport]``?

**Comment by timotheecour** [comment]
no idea, i just added `fixed in devel` because it works in devel; idk if it's 1 commit or more that fixes this

**Comment by narimiran** [comment]
> git bisect could help to identify it

I did `git bisect` on `version-0-19` branch and it spit out https://github.com/nim-lang/Nim/commit/41e2f6b7bc9666af7076ffd47006c8e813c1a854 as a first bad commit ;)

**Comment by timotheecour** [comment]
but that fixed a regression itself,
```
fix #9264 regression (#9265)
```

**Comment by narimiran** [comment]
> so would probably need to backport stuff like #9259 

You're right, backporting that fixed this issue.

I just pushed that to `version-0-19`, so the fix will be part of v0.19.4 — this can be closed now.
